src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtpositiontype/DebtPositionTypeProcessingService.java

```diff
@@ -0,0 +1,130 @@
+      IngestionFlowFile ingestionFlowFile, Path workingDirectory) {
+
+    List<DebtPositionTypeErrorDTO> errorList = new ArrayList<>();
+    DebtPositionTypeIngestionFlowFileResult ingestionFlowFileResult = new DebtPositionTypeIngestionFlowFileResult();
+    ingestionFlowFileResult.setDebtPositionTypeCodeList(new ArrayList<>());
+
+    Optional<Organization> organizationBroker = organizationService.getOrganizationById(
+        ingestionFlowFile.getOrganizationId());
+    Organization organization = organizationBroker.orElse(null);
+    Long brokerId = organization != null ? organization.getBrokerId() : null;
+    if (brokerId == null) {
+      log.error("Broker for organization id {} not found", ingestionFlowFile.getOrganizationId());
+      ingestionFlowFileResult.setErrorDescription("Broker not found");
+      return ingestionFlowFileResult;
+    }
+    ingestionFlowFileResult.setBrokerId(brokerId);
+    ingestionFlowFileResult.setBrokerFiscalCode(organization.getOrgFiscalCode());
+
+    process(iterator, readerException, ingestionFlowFileResult, ingestionFlowFile, errorList,
+        workingDirectory);
+    return ingestionFlowFileResult;
+  }
+
+  @Override
+  protected boolean consumeRow(long lineNumber,
+      DebtPositionTypeIngestionFlowFileDTO debtPositionTypeDTO,
+      DebtPositionTypeIngestionFlowFileResult ingestionFlowFileResult,
+      List<DebtPositionTypeErrorDTO> errorList, IngestionFlowFile ingestionFlowFile) {
+    try {
+      CollectionModelDebtPositionType existingDebtPosType = debtPositionTypeService.getByMainFields(
+          debtPositionTypeDTO.getDebtPositionTypeCode(),
+          ingestionFlowFileResult.getBrokerId(),
+          debtPositionTypeDTO.getOrgType(),
+          debtPositionTypeDTO.getMacroArea(),
+          debtPositionTypeDTO.getServiceType(),
+          debtPositionTypeDTO.getCollectingReason(),
+          debtPositionTypeDTO.getTaxonomyCode());
+
+      List<DebtPositionType> debtPositionTypeList = new ArrayList<>();
+      if (existingDebtPosType != null && existingDebtPosType.getEmbedded() != null && existingDebtPosType.getEmbedded().getDebtPositionTypes() != null) {
+        debtPositionTypeList = existingDebtPosType.getEmbedded().getDebtPositionTypes();
+      }
+
+      if (!debtPositionTypeList.isEmpty()) {
+        DebtPositionTypeErrorDTO error = new DebtPositionTypeErrorDTO(
+            ingestionFlowFile.getFileName(), debtPositionTypeDTO.getDebtPositionTypeCode(),
+            debtPositionTypeDTO.getBrokerCf(), lineNumber, "DEBT_POSITION_TYPE_ALREADY_EXISTS",
+            "Debt position type already exists");
+        errorList.add(error);
+        return false;
+      }
+
+      DebtPositionType debtPositionTypeCreated = debtPositionTypeService.createDebtPositionType(
+          debtPositionTypeMapper.map(debtPositionTypeDTO, ingestionFlowFileResult.getBrokerId()));
+      ingestionFlowFileResult.getDebtPositionTypeCodeList().add(debtPositionTypeCreated.getCode());
```

Comment:
if you don't plan to use it, let's avoid to store on ingestionFlowFileResult this...
Temporal will store this object on its database...